pong-game/game-service/app/game_service/consumers.py

```diff
@@ -0,0 +1,185 @@
+from channels.sessions import SessionMiddlewareStack
+from channels.layers import get_channel_layer
+from channels.testing import WebsocketCommunicator
+from .game_room import GameRoom
+
+load_dotenv()
+active_game_rooms = {}
+active_lobbies = {} #This has no methods cleaning it up yet. Need to clean when gamerooms terminate or if players leave a lobby
+logger = logging.getLogger(__name__)
+
+class GameConsumer(AsyncWebsocketConsumer):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.channel_layer = get_channel_layer()
+
+    async def connect(self):
+        logger.info(f"WebSocket connection attempt: {self.scope['path']}")
+        try:
+            await self.accept()
+        except Exception as e:
+            logger.error(f"WebSocket connection error: {e}")
+        await self.send(json.dumps({
+            "type": "notice",
+            "message": "Connection established"
+            }))
+
+    async def disconnect(self, code): #beef it up with socket closing and such
+        if hasattr(self, 'current_group'):
+            await self.channel_layer.group_discard(self.current_group, self.channel_name)
+
+    async def receive(self, text_data=None, bytes_data=None):
+        if text_data is None:
+            return
+        data = json.loads(text_data)
+        action = data.get("action")
+        player_id = data.get("id")
+
+        if action == "join_private_match":
+            await self.join_lobby(data["room_name"], player_id)
+        elif action == "create_private_match":
+            room_name = str(uuid.uuid4())
+            await self.create_private_lobby(room_name, player_id)
+        elif action == "ready_up":
+            await self.update_ready_status(data["room_name"], data["player_id"])
+        elif data['type'] == "player_input":
+            roomID = data['game_roomID']
+            if roomID in active_game_rooms:
+                game_room = active_game_rooms[roomID]
+                logger.info("Consumer: Received player input")
+                await game_room.receive_player_input(data['player_id'], data['input'])
+                logger.info("Consumer: Forwarded player input")
+            else:
+                await self.send(json.dumps({
+                       "type": "error",
+                       "message": f"Game room {data['game_roomID']} not found"
+                       }))
+
+    async def create_private_lobby(self, room_name, player_id):
+        active_lobbies[room_name] = {"players": [player_id]}
+        self.current_group = f"lobby_{room_name}"
+        await self.channel_layer.group_add(self.current_group, self.channel_name)
+        await self.send(json.dumps({
+            "type": "room_creation",
+            "message": f"Created Lobby {room_name}",
+            "room_name": room_name
+            }))
+
+    async def join_lobby(self, room_name, player_id):
+        if room_name not in active_lobbies:
```

Comment:
Harmonize How room_names are defined, either settle for "lobby_" + uid or just the uid alone